### ISSUE TITLE
Think about what we did wrong and how to continue

### ISSUE THREAD

**Issue by MarcWeber** [MarcWeber/vim-addon-manager#148]
Despite VAM works great we did something wrong - user estimations (2014-02):
[vundle#rc hits @ google](https://www.google.de/search?q=site:github.com+vundle%23rc)                   (approx 272.000 hits)
[vam#ActivateAddons hits @ google](https://www.google.de/search?q=site:github.com+vam%23ActivateAddons) (approx 800 hits)

ratio (google hits): 340
star ratio (github): 13
fork ratio: 9

Maybe I've been mistaken and nobody wants a managed pool of plugins, maybe we didn't care enough about README.md ... 

Anyway - time to think about how much additional time to spend on VAM ..

**Comment by rprs** [comment]
I must agree with RedX2501. I've spent the past two days comparing between vam and vundle. Easy of use is Vundle's killer feature when compared to vam. I've spent 90% of the time reading about vam and 10% on vundle. I'm still not so sure how to use vam (do I need to execute vam#update manually or does vam#activateAddons does it by itself?).

I'm still going to give vam a try, I just want to reiterate the importance of ease of use. I believe it is the reason behind whatsapp popularity.

**Comment by rprs** [comment]
After a few months of using vam, I must admit I was wrong. Vam is pretty awesome. Specially when setting a new machine: just turn on vim and you will get all your plugins automatically loaded. That's pretty slick!

I still think better instructions could improve ease of ramp up, but I'm a complete "vamer" now.

**Comment by nhooyr** [comment]
@rprs vim-plug can do that too. https://github.com/junegunn/vim-plug/wiki/faq

**Comment by MarcWeber** [comment]
I do no longer want to discuss 'which solution is good/bad' but whether it would be possible to share a common configuration so that you can switch implementations - because VAM is missing a concurrent installation/update system which could be added for "most plugins" which don't require viml hooks easily by putting plugins into a text file.

@rps "Activate" means "activate", "update" means update. Activate means "load the plugin and install if necessary". You can run install manually then activate in two phases if you want to look at the code first because in the end you're downloading "untrusted code from github". So adding "code review" from different parties would be nice have, but is out of scope for VAM and should be done on a different level.

@bluddy Having a window with *all* plugins means http://vam.mawercer.de/ -> huge list -> a lot of plugins are 10 years old and broken or outdated and does not help the user IMHO. We have name completion and command line completion. Creating such list is trivial because we have the dictionary with plugins ready. We could add a :ListAddons command, I agree.

AddonInfo shows:

```
  ===== by-name ===================================================================================================================
  Plugin: vim-addon-manager
  Script number: 2905
  Vim.org page: http://www.vim.org/scripts/script.php?script_id=2905
  Home page: https://github.com/MarcWeber/vim-addon-manager
  Source URL: git://github.com/MarcWeber/vim-addon-manager (type git)

````
We could work on adding description. Most value would be provided to users to see which plugins actually get used most often eventually or adding key values such as "got uninstalled within a short period of time" to help make a judgment about which plugins to try first.

**Comment by rps** [comment]
@rprs ^

**Comment by MarcWeber** [comment]
Long time no see.

vam was written because nothing better existed.
Its goals were to allow distributing plugins with dependencies.
And make loading them lazily easy while allowing the sources where to install
something to be flexible. Eg install YouCompleteMe from nixpks yet allowing to
augment with :VAMActivate github:name/repo or just name/repo.

The issues were: VimL was limited eg it was hard to implement parallel git pull
many. What's the right way to checkpoint your own setup etc.

VAM tried to be >latest only< + giving plugins a unique name so that
communication is easy while eventually allowing to overwrite details in special
situations.

Now I tried to describe how I want the future to look like:
Cause the requirements nix nix-home vs vim vs .. who is in charge to install
stuff and when eagerly lazily etc. But whatever you do you need some kind of
env which describes where to install what and how and where to find packages.
The when the using application could choose.

nixpkgs: declarative
nix/eclipse/vscode/vundle/
https://github.com/mason-org/mason.nvim

If you want to brain storm about the future I created a repository to discuss
a tool which works in more contexts. Eventually just comment on the first issue.
https://github.com/MarcWeber/crossy

VAM isn't dead. But even nixpkgs dropped it despite it was working well.
So consider it to be working but used by some view people only.

And about vim-pi -> best to find copy on github and reference them.
Now that Vim is dead (well it will never die but ..) most future plugins will
be new lua based anyway.

So what was mentioned in this thread like should I be usin :VAMUpdate .. yes/no depends. You could have used a shell command for i in .. git pull if you were ok with missing helptags or such. And I somewhat agree that VAM was little bit ego driven and went into great details and maybe was over engineered - but it got the job done. And the main code was not much and fast so it was good enough for me :-) And thanks for the honest discussion in this thread. It just tells me that communites are very important.
